### PR TITLE
Compiler: several require fixes

### PR DESCRIPTION
**Pull request by dustyo-O**
Fixes #
require module issues: 
- when you require a module giving it the name different from original, module provided still has original name.
- actually no modules provided to buildBemXjst
 
_added miripiruni to my fork_

### CODE REVIEW

File: lib/compiler.js [hunk 1]
```diff
@@ -150,24 +144,33 @@ Compiler.prototype.generate = function generate(code, options) {
...
-      deps.commonJSNames.length === 0 || (
+      deps.defineAsGlobal || deps.commonJSNames.length === 0 || (
         'if (typeof module==="object" && typeof module.exports==="object") {' +
-        'exp["' + exportName + '"] = buildBemXjst();' +
-        'exp["' + exportName + '"].libs = {};' + (
+        'var _libs = {};' + (
           deps.commonJSNames.map(function(dep) {
-            return 'exp["' + exportName + '"].libs["' + dep + '"] = ' +
+            var commonPath = deps.commonJS[dep];
+            return 'console.log("' + exportName + '");' +'_libs["' + dep + '"] = ' +
```
Comment (tadatuta) [inline lib/compiler.js]: ?

**Comment by dustyo-O** [comment]
why linters are not in the pre commit hook?

**Comment by tadatuta** [comment]
now specs on export are broken: https://travis-ci.org/bem/bem-xjst/jobs/285454882#L689

**Comment by miripiruni** [comment]
@dustyo-O @tadatuta the branch was updated by myself.

File: lib/compiler.js [hunk 2]
```diff
@@ -29,6 +29,7 @@ function getDeps(requires) {
     commonJSNames: [],
     ym: [],
     ymVars: [],
+    ymVarsHash: [],
```
Comment (dustyo-O) [inline lib/compiler.js]: '' looks better for me

**Pull request merged**